aten/src/ATen/native/mkldnn/Linear.cpp

```diff
@@ -24,22 +24,45 @@ Tensor mkldnn_linear(
 namespace at {
 namespace native {
 
+// Helper function for getting an ideep tensor out of an aten Tensor.
+// Note in case the aten Tensor is a dense tensor, make sure:
+//   1. the aten tensor is 2d, so that 3d*2d Linear is viewed as 2d*2d.
+//   2. the aten tensor is contiguous.
+inline ideep::tensor get_mkldnn_tensor(const at::Tensor& tensor) {
+  if (tensor.is_mkldnn()) {
+    return at::native::itensor_from_mkldnn(tensor);
+  } else {
+    auto x = tensor.contiguous().view({-1, tensor.size(tensor.dim() - 1)});
```

Comment:
you probably want to apply it only to X, not W. So I'd move it outside of this function (or make configurable)

Reply:
yes, good point. W doesn't need this logic. Fixed in `66a0ffcb`